### PR TITLE
build: bump cpu image to latest ubuntu 2004

### PR DESCRIPTION
**Pull request by hamidzr**
https://circleci.com/developer/images?imageType=machine

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by hamidzr** [comment]
https://app.circleci.com/pipelines/github/determined-ai/determined/51903/workflows/a4e26de5-b46b-46e5-a7a3-5f2cf673593a/jobs/2316513/artifacts

not sure how to update the image for the `..package..` job or what it's currently using

**Review by mackrorysd — APPROVED** [review]

File: .circleci/real_config.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ parameters:
     default: determinedai/cimg-base:latest
   machine-image:
     type: string
-    default: ubuntu-2004:202201-02
+    default: ubuntu-2004:2024.01.1
   gpu-machine-image:
     type: string
     default: ubuntu-2004-cuda-11.2:202103-01
```
Comment (djanicekpach) [inline .circleci/real_config.yml:37]: This GPU image also looks a bit older. Do you know if it ok or does it need an upgrade too? 
Edit - probably good just to upgrade now 
https://discuss.circleci.com/t/linux-cuda-deprecation-and-image-policy/48568

**Review by djanicekpach — APPROVED** [review]
Looks good, one small comment

File: .circleci/real_config.yml (hunk 1 above)
Comment (hamidzr) [reply .circleci/real_config.yml:37]: I agree I [considered](https://github.com/determined-ai/determined/pull/8940/commits/4592304341e8bd424cbc8b2023bb3a83ae0a343b) it but I wasn't sure about the side effects and I want get these changes in asap

Comment (mackrorysd) [reply .circleci/real_config.yml:37]: @djanicekpach There was this discussion on Slack: https://hpe-aiatscale.slack.com/archives/C04C9JXB1C2/p1709567985738729?thread_ts=1709565408.754739&cid=C04C9JXB1C2. Aside from that specific implication, which I may look into, I'll start a branch with a bulk upgrade to a more recent Ubuntu version and see how it goes.

**Pull request merged**